src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1003,6 +1003,14 @@ parameters[i].RefKind switch
 
         private void EmitArrayElementLoad(BoundArrayAccess arrayAccess, bool used)
         {
+            if (arrayAccess.IsRef)
+            {
+                Debug.Assert(!used);
+                EmitArrayElementAddress(arrayAccess, AddressKind.Writeable);
+                EmitPopIfUnused(used);
```

Comment:
>EmitPopIfUnused(used); [](http://example.com/codeflow?start=16&length=22)

I think we cannot leave reference on the stack when a value is expected. Therefore, we should always pop and throw if ```used``` is set